memcache/selector.go

```diff
@@ -31,8 +32,8 @@ import (
 type ServerSelector interface {
 	// PickServer returns the server address that a given item
 	// should be shared onto.
-	PickServer(key string) (net.Addr, error)
-	Each(func(net.Addr) error) error
+	PickServer(ctx context.Context, key string) (net.Addr, error)
```

Comment:
1. This is breaking change. I doubt it can be accepted.
2. Wonder what is the use case to add context for these methods. You can have background goroutine which updates servers list if needed. Current implementation also indicates that context is not needed.